runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -83,19 +87,30 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** {@link DataflowExecutionContext} for use in streaming mode. */
+/**
+ * {@link DataflowExecutionContext} for use in streaming mode. Contains cached readers and Beam
+ * state pertaining to a processing its owning computation. Can be reused across processing
+ * different WorkItems for the same computation.
+ */
 @SuppressWarnings({
   "nullness" // TODO(https://github.com/apache/beam/issues/20497)
```

Comment:
Compiler will complain with under initialized error will update to prevent both

Reply:
there are quite a bit of nullability issues not just with work nullability 

added a comment to deal with this a follow up PR

Reply:
I'm wondering if the separate state via biulder and then accessor instead of just local variables will lead to performance regression since this is on the hot path.

Since we don't actually clear the workScopedState between work items it seems that the null check will only help before start is ever called and then the additional machinery will be a no-op but more expensive.

Can you revert back to existing local variables for now? Perhaps we can improve nullability/correctness verification while merging this and the ComputationWorkExecutor and do a perf comparison.  I don't want to make this big change riskier.

Reply:
done sgtm!